### PR TITLE
Redirected users to registration page if they are not registered yet

### PR DESCRIPTION
**Pull request by apluscs**

### CODE REVIEW

**Review by meagancheese — CHANGES_REQUESTED** [review]
Is getUserIfRegistered() the best name for this method? Maybe it can't be helped because that is its main purpose on the registration page, but I think a separate method in a different file (like supplyLogoutLink is) would be better.

File: src/main/webapp/profile.js [hunk 1]
```diff
@@ -13,8 +13,9 @@
 // limitations under the License.
 
 function onProfileLoad() {
-  supplyLogoutLink();  
-  getUserIfRegistered().then((person) => {
+  const loginInfo = getLoginInfo();
+  loginInfo.then(supplyLogoutLinkOrRedirectHome);  
+  loginInfo.then(getUserOrRedirectRegistration).then((person) => {
```
Comment (meagancheese) [inline src/main/webapp/profile.js:18]: Does this function properly when the function returns nothing?

**Review by meagancheese — APPROVED** [review]
One question, but otherwise LGTM.

**Review by GrantComm — APPROVED** [review]

File: src/main/webapp/authentication.js [hunk 2]
```diff
@@ -13,12 +13,27 @@
...
+// Returns a Person if they registered in the past. If not, redirect to  
+// registration page.
+function getUserOrRedirectRegistration(loginInfo){
+  return fetch(`/person?email=${loginInfo.email}`)
+    .then(response => {
+      if (response.redirected) {
+        window.location.href = response.url;
+        return;
+      }
+      return response.json();
```
Comment (jalexanderqed) [inline src/main/webapp/authentication.js:37]: Please confirm that this works manually. As we experienced during the meeting, `response.redirected` may not have the value you expect.

**Review by jalexanderqed — DISMISSED** [review]
Feel free to send to Alex for final approval since I'm not sure I will get to it today.

File: src/main/webapp/authentication.js (hunk 2 above)
Comment (apluscs) [reply src/main/webapp/authentication.js:37]: lt works when l checked it manually

**Review by acmcarther — APPROVED** [review]

**Pull request merged**